controllers/datamovement_controller.go

```diff
@@ -89,314 +90,275 @@ func (r *DataMovementReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// Check if the object is being deleted. Deletion is coordinated around the sub-resources
-	// created or modified as part of data movement.
 	if !dm.GetDeletionTimestamp().IsZero() {
-		log.Info("Starting delete operation")
 
-		if !controllerutil.ContainsFinalizer(dm, finalizer) {
-			return ctrl.Result{}, nil
-		}
+		// TODO: Cancel/Abort the operation from Blake
 
-		teardownFns := []func(context.Context, *nnfv1alpha1.NnfDataMovement) (*ctrl.Result, error){
-			r.teardownRsyncJob,
-			r.teardownLustreJob,
-		}
+		if controllerutil.ContainsFinalizer(dm, finalizer) {
+			controllerutil.RemoveFinalizer(dm, finalizer)
 
-		for _, teardownFn := range teardownFns {
-			result, err := teardownFn(ctx, dm)
-			log.Info("Teardown", "Result", result, "Error", err)
-			if err != nil {
+			if err := r.Update(ctx, dm); err != nil {
 				return ctrl.Result{}, err
-			} else if result != nil {
-				return *result, nil
 			}
 		}
 
-		controllerutil.RemoveFinalizer(dm, finalizer)
+		return ctrl.Result{}, nil
+	}
+
+	if !controllerutil.ContainsFinalizer(dm, finalizer) {
```

Comment:
If you tried to remove all the finalizers, wouldn't this just add it right back? Wouldn't that make the dwsutil purge functionality not so useful?